### DISCUSSION TITLE
Any special techniques for puzzles with letters removed?

### DISCUSSION THREAD

**Discussion by DKMiller71** [viresh-ratnakar/exolve#76] [Q&A] [answered]
Not saying it's necessarily needed, but: are there any options for tracking words/letters for puzzles that have an element of removing a letter or letters before adding it to the grid?

**Comment by viresh-ratnakar** [comment]
Not at the moment.

However, this is done commonly enough that I can consider adding some support. If you have ideas on what this feature can look like, please add more thoughts here. Thanks!

**Reply by DKMiller71** [reply]
Yeah, I've been trying my first cryptic with missing letters and was just thinking for the player it would be nice to have them able to enter their idea for the word - which they might not know how to place yet - kind of like you do for unmarked clues that provide the "fill in button".

In my puzzle, the dropped letters are part of a final extraction, so I was wondering whether it was useful/practical to have it display the difference between the word the player specified and the letters actually entered into the grid.  For example, the clue solves to ANSWER, but the grid only takes ANWER, then S would be the extra letter.

That said, I'm not that familiar with all the different ways dropped letters might get used, so I'm not sure I'd have a good grasp on the possibilities.

**Reply by DKMiller71** [reply]
One of the people working on my cryptic with missing letters said the following, if that helps.  It kind of aligns with what I had said:

> I used the filler page, but also needed to print it out to write answers I couldn't enter just yet. 

So, sounds like having a place to track words you think you've solved may be an appreciated thing.

**Comment by viresh-ratnakar** [comment, marked as answer]
I've added a way to do this now, with v1.29. Here's the relevant bit from the README file:

https://github.com/viresh-ratnakar/exolve/blob/master/README.md#forcing-the-display-of-placeholder-blanks

### Forcing the display of "placeholder blanks"

Placeholder blanks normally get displayed only in front of "orphan" clues whose
light locations are not provided to the solver. However, you can force a
placeholder blank next to any clue by following it with one or more underscores.
If you're providing annotations, place the undescrores before the annotations.
Examples:
```
  exolve-across:
    5 This clue will get placeholder blanks (4) _
    6 Here we're specifying that there should be 7 blanks, regardless of enum (4) _______
    7 The underscores can have intervening spaces and can be followed by annos (8) _ _ _ Some anno.
```

If you place just one underscore, then the actual displayed size of the blank
will be determined using the enum. If you place more than one underscore, then
the displayed size of the blank will equal the number of underscores that you
have provided.

Just like the placeholder blanks that appear in orphan clues, these forced
placeholder blanks will also be accompanied by "copy-placeholder buttons"
(that looks like [⇲]), unless disabled by specifying
`exolve-option: hide-copy-placeholder-buttons`.

**Comment by DKMiller71** [comment]
Looks good. If you want see it action:
https://dkmiller71.github.io/DarrensPuzzles/puzzles/xword/ThanksgivingLeftovers.html
